### PR TITLE
Fix blocking calls preventing streaming

### PR DESCRIPTION
**Pull request by collindutter**

### CODE REVIEW

File: griptapecli/core/skatepark.py [hunk 1]
```diff
@@ -163,3 +161,21 @@ def _validate_files(structure: Structure):
...
         raise HTTPException(status_code=400, detail="requirements.txt does not exist")
+
+
+def _check_run_process(run_process: RunProcess) -> RunProcess:
+    process = run_process.process
+
+    if process is not None:
+        return_code = process.poll()
+        if return_code is not None:
+            stdout, stderr = run_process.process.communicate()
```
Comment (collindutter) [inline griptapecli/core/skatepark.py:172]: This is the key change. `communicate` blocks, `poll` does not.

**Review by vasinov — APPROVED** [review]

File: griptapecli/commands/skatepark.py [hunk 2]
```diff
@@ -65,9 +65,7 @@ def start(
     port: int,
 ) -> None:
     """Starts the Griptape server."""
-    uvicorn.run(
-        "griptapecli.core.skatepark:app", host=host, port=port, reload=False, workers=4
-    )
+    uvicorn.run("griptapecli.core.skatepark:app", host=host, port=port, reload=False)
```
Comment (SavagePencil) [inline griptapecli/commands/skatepark.py:68]: what does `workers` default to?

File: griptapecli/core/skatepark.py [hunk 3]
```diff
@@ -101,7 +103,7 @@ def list_structure_runs(structure_id: str) -> list[Run]:
     ]
 
 
-@app.patch("/api/runs/{run_id}")
+@app.patch("/api/structure-runs/{run_id}")
```
Comment (SavagePencil) [inline griptapecli/core/skatepark.py:106]: did we change this in Cloud?

**Review by SavagePencil — APPROVED** [review]
minor Qs

File: griptapecli/core/skatepark.py (hunk 3 above)
Comment (collindutter) [reply griptapecli/core/skatepark.py:106]: Yes

File: griptapecli/commands/skatepark.py (hunk 2 above)
Comment (collindutter) [reply griptapecli/commands/skatepark.py:68]: `1` but I will explicitly set it in an upcoming PR.

**Pull request merged**